dali/operators/image/resize/resize_attr_test.cc

```diff
@@ -0,0 +1,382 @@
+      CHECK_PARAMS(attr.params_[1], { { 40, 400, 300 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+
+    {
+      spec.AddArg("resize_z", 600.0f);
+
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+      CHECK_PARAMS(attr.params_[0], { { 600, 300, 400 }, { 0, 0, 0 }, { 1536, 768, 1024 } });
+      CHECK_PARAMS(attr.params_[1], { { 40, 400, 300 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+  }
+}
+
+TEST(ResizeAttr, Resize3DNotSmaller) {
+  ArgumentWorkspace ws;
+  {
+    OpSpec spec("Resize");
+    spec.AddArg("resize_x", 600.0f);
+    spec.AddArg("resize_y", 480.0f);
+    spec.AddArg("mode", "not_smaller");
+    TensorListShape<> shape = {{
+      TensorShape<3>{ 1536, 768, 1024 },
+      TensorShape<3>{ 32, 320, 240 }
+    }};
+    spec.AddArg("batch_size", shape.num_samples());
+
+    {
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+      CHECK_PARAMS(attr.params_[0], { { 960, 480, 640 }, { 0, 0, 0 }, { 1536, 768, 1024 } });
+      CHECK_PARAMS(attr.params_[1], { { 80, 800, 600 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+
+    {
+      spec.AddArg("resize_z", 160.0f);
+
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+      CHECK_PARAMS(attr.params_[0], { { 960, 480, 640 }, { 0, 0, 0 }, { 1536, 768, 1024 } });
+      CHECK_PARAMS(attr.params_[1], { { 160, 1600, 1200 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+
+
+    {
+      spec.SetArg("max_size", vector<float>{720.0f});
+
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+      CHECK_PARAMS(attr.params_[0], { { 720, 360, 480 }, { 0, 0, 0 }, { 1536, 768, 1024 } });
+      CHECK_PARAMS(attr.params_[1], { { 72, 720, 540 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+
+    {
+      spec.SetArg("max_size", vector<float>{720.0f, 384.0f, 400.0f});
+
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "DHW");
+
+      CHECK_PARAMS(attr.params_[0], { { 600, 300, 400 }, { 0, 0, 0 }, { 1536, 768, 1024 } });
+      CHECK_PARAMS(attr.params_[1], { { 38, 384, 288 }, { 0, 0, 0 }, { 32, 320, 240 } });
+    }
+  }
+}
+
+
+TEST(ResizeAttr, ResizeShorter) {
+  ArgumentWorkspace ws;
+  {
+    OpSpec spec("Resize");
+    spec.AddArg("resize_shorter", 600.0f);
+    TensorListShape<> shape = {{
+      TensorShape<4>{ 20, 400, 800, 3 },
+      TensorShape<4>{ 30, 500, 250, 3 }
+    }};
+    spec.AddArg("batch_size", shape.num_samples());
+
+    {
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "FHWC");
+
+      CHECK_PARAMS(attr.params_[0], { { 600, 1200 }, { 0, 0 }, { 400, 800 } });
+      CHECK_PARAMS(attr.params_[1], { { 1200, 600 }, { 0, 0 }, { 500, 250 } });
+    }
+
+    {
+      spec.SetArg("max_size", vector<float>{800.0f, 1000.0f});
+
+      ResizeAttr attr;
+      attr.PrepareResizeParams(spec, ws, shape, "FHWC");
+
+      CHECK_PARAMS(attr.params_[0], { { 500, 1000 }, { 0, 0 }, { 400, 800 } });
```

Comment:
I don't know what we should do in such case. Definitely it doesn't follow `resize_shorter=600`.

Reply:
That's what max_size is for and how it works now (except that until now we had max_size uniform for all axes). It works as intended and passes the legacy tests.